public/components/trace_analytics/components/traces/span_detail_panel.tsx

```diff
@@ -187,14 +187,27 @@ export function SpanDetailPanel(props: {
 
   const [currentSpan, setCurrentSpan] = useState('');
 
+  const spanContainerRef = useRef<HTMLDivElement | null>(null);
+  const prevScrollPositionRef = useRef<number | null>(null);
+
+  useEffect(() => {
+    if (spanContainerRef.current) {
+      setTimeout(() => {
+        spanContainerRef.current.scrollTop = prevScrollPositionRef.current || 0;
+      }, 0);
```

Comment:
Question: why do we need `setTimeout` with delay 0 here?

Reply:
Without it, closing the fly-out will reset the scroll bar. I believe this happens because the fly-out table closing behavior is handled in span_detail_table and it needs a delay for the use-effect.